From: Stefan Wahren <wahrenst@gmx.net>
Date: Sun, 8 Dec 2019 14:45:14 +0100
Subject: Revert "mmc: sdhci: Mask "spurious" interrupts"

This reverts commit adac895210b88770190e9a501462f2100ab40ac8, because this
issue is already addressed by commit 19ec6bb80290e496021488084944b77f03a87dd7.

Signed-off-by: Stefan Wahren <wahrenst@gmx.net>
---
 drivers/mmc/host/sdhci.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/drivers/mmc/host/sdhci.c b/drivers/mmc/host/sdhci.c
index 050f7c81bf2e..deafcc56adee 100644
--- a/drivers/mmc/host/sdhci.c
+++ b/drivers/mmc/host/sdhci.c
@@ -3130,10 +3130,6 @@ static irqreturn_t sdhci_irq(int irq, void *dev_id)
 			result = IRQ_WAKE_THREAD;
 		}
 
-		if ((intmask & SDHCI_INT_DATA_END) && !host->data &&
-		    host->cmd && (host->cmd == host->cmd->mrq->stop))
-			intmask &= ~SDHCI_INT_DATA_END;
-
 		if (intmask & SDHCI_INT_CMD_MASK)
 			sdhci_cmd_irq(host, intmask & SDHCI_INT_CMD_MASK, &intmask);
 
